apps/web/src/components/WalletAddressInput.tsx

```diff
@@ -0,0 +1,252 @@
+import { useEffect, useState, useRef } from "react";
+import clsx from "clsx";
+import * as ethers from "ethers";
+import { FiClipboard } from "react-icons/fi";
+import { IoCloseCircle } from "react-icons/io5";
+import { fromAddress } from "@defichain/jellyfish-address";
+import { useNetworkEnvironmentContext } from "@contexts/NetworkEnvironmentContext";
+import useResponsive from "@hooks/useResponsive";
+import useAutoResizeTextArea from "@hooks/useAutoResizeTextArea";
+import { Network } from "types";
+import Tooltip from "./commons/Tooltip";
+import EnvironmentNetworkSwitch from "./EnvironmentNetworkSwitch";
+
+interface Props {
+  blockchain: Network;
+  label: string;
+  disabled?: boolean;
+}
+
+const blockchainNameMap: Record<Network, string> = {
+  DeFiChain: "DeFiChain",
+  Ethereum: "ERC20",
+};
+
+export default function WalletAddressInput({
+  blockchain,
+  label,
+  disabled = false,
+}: Props): JSX.Element {
+  const [addressInput, setAddressInput] = useState<string>("");
+  const [isValidAddress, setIsValidAddress] = useState(false);
+  const [isFocused, setIsFocused] = useState(false);
+  const textAreaRef = useRef<HTMLTextAreaElement>(null);
+  const [placeholder, setPlaceholder] = useState<string>("");
+  const [error, setError] = useState({ message: "", isError: false });
+  const [copiedFromClipboard, setCopiedFromClipboard] = useState(false);
+
+  const { networkEnv, networkEnvDisplayName } = useNetworkEnvironmentContext();
+  const { isSm } = useResponsive();
+  useAutoResizeTextArea(textAreaRef.current, [addressInput, placeholder]);
+
+  const validateAddressInput = (input: string): void => {
+    let isValid = false;
+    if (blockchain === "Ethereum") {
+      isValid = ethers.utils.isAddress(input);
+    } else {
+      const decodedAddress = fromAddress(input, networkEnv);
+      isValid = decodedAddress !== undefined;
+    }
+    setIsValidAddress(isValid);
+  };
+
+  const handlePasteBtnClick = async () => {
+    if (disabled) return;
+    const copiedText = await navigator.clipboard.readText();
+    if (copiedText) {
+      setAddressInput(copiedText);
+      setCopiedFromClipboard(true);
+    }
+  };
+
+  const handleFocusWithCursor = () => {
+    setIsFocused(true);
+    setTimeout(() => {
+      // Only added timeout for ref's unexplained delay
+      const textArea = textAreaRef.current;
+      const cursorPosition = addressInput.length;
+      if (textArea) {
+        textArea.setSelectionRange(cursorPosition, cursorPosition);
+        textArea.focus();
+      }
+    }, 1);
```

Comment:
does it work if we change to this? this is still delaying, by one tick
```suggestion
    }, 0);
```